src/palladio/NodeParameter.cpp

```diff
@@ -169,63 +168,29 @@ std::filesystem::path getRPK(const OP_Node* node, fpreal t) {
 	return s.toStdString();
 }
 
-void buildRuleFileMenu(void* data, PRM_Name* theMenu, int theMaxSize, const PRM_SpareData*, const PRM_Parm* parm) {
-	const auto* node = static_cast<SOPAssign*>(data);
-	const auto& prtCtx = node->getPRTCtx();
-
-	const fpreal now = CHgetEvalTime();
-	const std::filesystem::path rpk = getRPK(node, now);
-
-	ResolveMapSPtr resolveMap = prtCtx->getResolveMap(rpk);
-	if (!resolveMap) {
-		theMenu[0].setToken(nullptr);
-		return;
-	}
-
-	std::vector<std::pair<std::wstring, std::wstring>> cgbs; // key -> uri
-	getCGBs(resolveMap, cgbs);
-
-	const size_t limit = std::min<size_t>(cgbs.size(), static_cast<size_t>(theMaxSize));
-	for (size_t ri = 0; ri < limit; ri++) {
-		std::string tok = toOSNarrowFromUTF16(cgbs[ri].first);
-		theMenu[ri].setTokenAndLabel(tok.c_str(), tok.c_str());
-	}
-	theMenu[limit].setTokenAndLabel(nullptr, nullptr); // need a null terminator
-}
-
-std::wstring getRuleFile(const OP_Node* node, fpreal t) {
-	UT_String s;
-	node->evalString(s, RULE_FILE.getToken(), 0, t);
-	return toUTF16FromOSNarrow(s.toStdString());
-}
-
-void setRuleFile(OP_Node* node, const std::wstring& ruleFile, fpreal t) {
-	const UT_String val(toOSNarrowFromUTF16(ruleFile));
-	node->setString(val, CH_STRING_LITERAL, RULE_FILE.getToken(), 0, t);
-}
-
 void buildStyleMenu(void* data, PRM_Name* theMenu, int theMaxSize, const PRM_SpareData*, const PRM_Parm*) {
 	const auto* node = static_cast<SOPAssign*>(data);
 	const PRTContextUPtr& prtCtx = node->getPRTCtx();
 
 	const fpreal now = CHgetEvalTime();
 	const std::filesystem::path rpk = getRPK(node, now);
-	const std::wstring ruleFile = getRuleFile(node, now);
 
 	ResolveMapSPtr resolveMap = prtCtx->getResolveMap(rpk);
 	if (!resolveMap) {
 		theMenu[0].setTokenAndLabel(nullptr, nullptr);
 		return;
 	}
 
-	const wchar_t* cgbURI = resolveMap->getString(ruleFile.c_str());
-	if (cgbURI == nullptr) {
+	std::vector<std::pair<std::wstring, std::wstring>> cgbs; // key -> uri
+	getCGBs(resolveMap, cgbs);
```

Comment:
consider introducing a similar helper function `getCGB` which just returns the one CGB URI (basically the first item returned from `getCGBs`). The original helper was written before we locked down RPKs to only one CGB.